identity-apps-core/apps/authentication-portal/src/main/webapp/includes/init-url.jsp

```diff
@@ -43,6 +44,11 @@
             spAppName = (String) request.getAttribute(SERVICE_PROVIDER_NAME_SHORT);
         }
 
+        spAppId = request.getParameter(SERVICE_PROVIDER_ID_SHORT);
+        if (StringUtils.isBlank(spAppId) && StringUtils.isNotBlank((String) request.getAttribute(SERVICE_PROVIDER_ID_SHORT))) {
```

Comment:
Wouldn't this break if `request.getAttribute(SERVICE_PROVIDER_ID_SHORT)` is null?

Reply:
It will fall back to default branding as expected.